index.js

```diff
@@ -111,13 +110,35 @@ const isDnaUnique = (_DnaList = [], _dna = []) => {
   return foundDna == undefined ? true : false;
 };
 
+const getRandomRarity = (_rarityOptions) => {
+  let randomPercent = Math.random() * 100;
+  let percentCount = 0;
+
+  for (let i = 0; i <= _rarityOptions.length; i++) {
+    percentCount += _rarityOptions[i].percent;
+    if (percentCount >= randomPercent) {
+      console.log(`use random rarity ${_rarityOptions[i].id}`)
+      return _rarityOptions[i].id;
+    }
+  }
+  return _rarityOptions[0].id;
+}
+
 // create a dna based on the available layers for the given rarity
 // use a random part for each layer
 const createDna = (_layers, _rarity) => {
   let randNum = [];
+  let _rarityWeight = rarityWeights.find(rw => rw.value === _rarity);
   _layers.forEach((layer) => {
-    let num = Math.floor(Math.random() * layer.elements[_rarity].length);
-    randNum.push(num);
+    let num = Math.floor(Math.random() * layer.elementIdsForRarity[_rarity].length);
+    if (_rarityWeight && _rarityWeight.layerPercent[layer.id]) {
```

Comment:
if a `_rarityWeight.layerPercent` object is available for the given layer, then run the new `getRandomRarity` function picking a rarity based on provided options and chances, instead of enforcing the layer to be of a certain rarity